lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h

```diff
@@ -0,0 +1,484 @@
+}
+SPIRV_DEF_NAMEMAP(LinkageType, SPIRVLinkageTypeNameMap)
+
+template<> inline void
+SPIRVMap<AccessQualifier, std::string>::init() {
+  add(AccessQualifierReadOnly, "ReadOnly");
+  add(AccessQualifierWriteOnly, "WriteOnly");
+  add(AccessQualifierReadWrite, "ReadWrite");
+}
+SPIRV_DEF_NAMEMAP(AccessQualifier, SPIRVAccessQualifierNameMap)
+
+template<> inline void
+SPIRVMap<FunctionParameterAttribute, std::string>::init() {
+  add(FunctionParameterAttributeZext, "Zext");
+  add(FunctionParameterAttributeSext, "Sext");
+  add(FunctionParameterAttributeByVal, "ByVal");
+  add(FunctionParameterAttributeSret, "Sret");
+  add(FunctionParameterAttributeNoAlias, "NoAlias");
+  add(FunctionParameterAttributeNoCapture, "NoCapture");
+  add(FunctionParameterAttributeNoWrite, "NoWrite");
+  add(FunctionParameterAttributeNoReadWrite, "NoReadWrite");
+}
+SPIRV_DEF_NAMEMAP(FunctionParameterAttribute, SPIRVFunctionParameterAttributeNameMap)
+
+template<> inline void
+SPIRVMap<Decoration, std::string>::init() {
+  add(DecorationRelaxedPrecision, "RelaxedPrecision");
+  add(DecorationSpecId, "SpecId");
+  add(DecorationBlock, "Block");
+  add(DecorationBufferBlock, "BufferBlock");
+  add(DecorationRowMajor, "RowMajor");
+  add(DecorationColMajor, "ColMajor");
+  add(DecorationArrayStride, "ArrayStride");
+  add(DecorationMatrixStride, "MatrixStride");
+  add(DecorationGLSLShared, "GLSLShared");
+  add(DecorationGLSLPacked, "GLSLPacked");
+  add(DecorationCPacked, "CPacked");
+  add(DecorationBuiltIn, "BuiltIn");
+  add(DecorationNoPerspective, "NoPerspective");
+  add(DecorationFlat, "Flat");
+  add(DecorationPatch, "Patch");
+  add(DecorationCentroid, "Centroid");
+  add(DecorationSample, "Sample");
+  add(DecorationInvariant, "Invariant");
+  add(DecorationRestrict, "Restrict");
+  add(DecorationAliased, "Aliased");
+  add(DecorationVolatile, "Volatile");
+  add(DecorationConstant, "Constant");
+  add(DecorationCoherent, "Coherent");
+  add(DecorationNonWritable, "NonWritable");
+  add(DecorationNonReadable, "NonReadable");
+  add(DecorationUniform, "Uniform");
+  add(DecorationSaturatedConversion, "SaturatedConversion");
+  add(DecorationStream, "Stream");
+  add(DecorationLocation, "Location");
+  add(DecorationComponent, "Component");
+  add(DecorationIndex, "Index");
+  add(DecorationBinding, "Binding");
+  add(DecorationDescriptorSet, "DescriptorSet");
+  add(DecorationOffset, "Offset");
+  add(DecorationXfbBuffer, "XfbBuffer");
+  add(DecorationXfbStride, "XfbStride");
+  add(DecorationFuncParamAttr, "FuncParamAttr");
+  add(DecorationFPRoundingMode, "FPRoundingMode");
+  add(DecorationFPFastMathMode, "FPFastMathMode");
+  add(DecorationLinkageAttributes, "LinkageAttributes");
+  add(DecorationNoContraction, "NoContraction");
+  add(DecorationInputAttachmentIndex, "InputAttachmentIndex");
+  add(DecorationAlignment, "Alignment");
+}
+SPIRV_DEF_NAMEMAP(Decoration, SPIRVDecorationNameMap)
+
+template<> inline void
+SPIRVMap<BuiltIn, std::string>::init() {
```

Comment:
These built-in names should have "BuiltIn" prefix per this document: https://github.com/KhronosGroup/SPIRV-LLVM/blob/khronos/spirv-3.6.1/docs/SPIRVRepresentationInLLVM.rst#id9